release/tasks/run-tests.ts

```diff
@@ -0,0 +1,105 @@
+import * as path from 'path';
+
+import { Package, Context } from '../@types/custom';
+import { Subscriber, Observable } from 'rxjs';
+import { debug, execa, execWithRetry, packageTask } from '../lib/utils';
+
+
+/**
+ * Fully tests a package
+ */
+const testPackage = async (pkg: Package) => {
+    // Package could be a configuration and nothing needs to be tested
+    if (!pkg.content.scripts || !pkg.content.scripts.test) {
+        return;
+    }
+
+    // npm run lint && npm run build && npm run test-only
+    const yarnTest = 'yarn test';
+
+    await execWithRetry(yarnTest, { cwd: path.dirname(pkg.path) });
+
+    pkg.built = true;
+    pkg.tested = true;
+};
+
+const buildPackages = async (pkgNames: string[], pkgs: Map<string, Package>) => {
+    for (const pkgName of pkgNames) {
+        const pkg = pkgs.get(pkgName)!;
+        const command = 'yarn build';
+
+        if (!pkg.built) {
+            await execa(command, { cwd: path.dirname(pkg.path) });
+            pkg.built = true;
+        }
+    }
+};
+
+const testPackages = async (packages: Map<string, Package>, observer: Subscriber<{}>) => {
+    let tested = 0;
+
+    for (const [, pkg] of packages) {
+        tested++;
+        if (pkg.updated) {
+            const message = `Building dependencies for ${pkg.name}`;
+
+            observer.next(message);
+            debug(message);
+
+            try {
+                /**
+                 * Build all the depedencies manually because there are some packages that need
+                 * the assets so `tsc -b` is not reliable.
+                 */
+                await buildPackages(pkg.references, packages);
+            } catch (e) {
+                observer.error(`Build failed for package ${pkg.name} dependencies`);
```

Comment:
I think the `observer.error`  stops the execution but now I'm not sure :/

Reply:
Yup, [`error` means completion](http://reactivex.io/documentation/contract.html) from what I read:

> An Observable may make zero or more OnNext notifications, each representing a single emitted item, and it may then follow those emission notifications by either an **OnCompleted or an OnError notification, but not both.**